integration-tests/messages/addIPFSMessage.test.ts

```diff
@@ -107,10 +107,24 @@ describe("Add Offchain Message", function () {
         }
     });
 
+    it("should successfully add onchain message", async function () {
+        const f = ExtrinsicHelper.addOnChainMessage(keys, dummySchemaId, "0xdeadbeef");
+        const [event] = await f.fundAndSend();
+
+        assert.notEqual(event, undefined, "should have returned a MessagesStored event");
+        if (event && f.api.events.messages.MessagesStored.is(event)) {
+            messageBlockNumber = event.data.blockNumber;
+            assert.deepEqual(event.data.schemaId, dummySchemaId, 'schema ids should be equal');
+            assert.notEqual(event.data.blockNumber, undefined, 'should have a block number');
+            assert.equal(event.data.count.toNumber(), 1, "message count should be 1");
+        }
+    });
+
```

Comment:
Nice. Although, this tests that a `MessageStored` event was generated by the extrinsic, but doesn't test that the message was successfully stored on-chain. Perhaps a second test to retrieve on-chain messages from that block & verify the content? (see ipfs retrieval test below)